lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -654,6 +654,24 @@ protected function getPreAlterTableIndexForeignKeySQL(TableDiff $diff)
             }
         }
 
+        // handle columns that are removed from changed indexes
+        foreach ($diff->changedIndexes as $chgKey => $chgIndex) {
+            // Dropping primary keys requires to unset autoincrement attribute on the particular column first
+            if ($chgIndex->isPrimary() && $diff->fromTable instanceof Table) {
+
+                foreach ($diff->fromTable->getIndex($chgIndex->getName())->getColumns() as $columnName) {
+                    $column = $diff->fromTable->getColumn($columnName);
+
+                    if ($column->getAutoincrement() === true && in_array($columnName, $chgIndex->getColumns()) === false) {
```

Comment:
Thinking out loud: is `in_array` a good idea here? `Index->hasColumnAtPosition` is much more complex.
I could either add a `indexHasColumn($index, $columnName)` private function mimicking the index function or could add an `hasColumn` function to `Index` (similar to `Table`) or allow `Index->hasColumnAtPosition` to allow `null` as position for `any`.

OR it is ok as it is ;)